grakn-graql/src/test/java/ai/grakn/graql/internal/reasoner/QueryValidityTest.java

```diff
@@ -0,0 +1,161 @@
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ */
+
+
+package ai.grakn.graql.internal.reasoner;
+
+import ai.grakn.exception.GraqlQueryException;
+import ai.grakn.graql.GetQuery;
+import ai.grakn.graql.QueryBuilder;
+import ai.grakn.test.rule.SampleKBContext;
+import org.junit.ClassRule;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.hamcrest.Matchers.empty;
+import static org.junit.Assert.assertThat;
+
+public class QueryValidityTest {
+
+    @Rule
+    public final ExpectedException expectedException = ExpectedException.none();
+
+    @ClassRule
+    public static final SampleKBContext testContext = SampleKBContext.load("ruleApplicabilityTest.gql");
+
+    @Test
+    public void whenQueryingForInexistentConceptId_emptyResultReturned(){
+        QueryBuilder qb = testContext.tx().graql().infer(true);
+        String queryString = "match $x id 'V123'; $y id 'V456'; ($x, $y); get;";
+        assertThat(qb.<GetQuery>parse(queryString).execute(), empty());
+    }
+
+    @Test
+    public void whenQueryingForInexistentEntityTypeId_emptyResultReturned(){
+        QueryBuilder qb = testContext.tx().graql().infer(true);
+        String queryString = "match $x isa $type; $type id 'V123'; get;";
+        assertThat(qb.<GetQuery>parse(queryString).execute(), empty());
+    }
+
+    @Test
+    public void whenQueryingForInexistentRelationTypeId_emptyResultReturned(){
+        QueryBuilder qb = testContext.tx().graql().infer(true);
+        String queryString = "match ($x, $y) isa $type; $type id 'V123'; get;";
+        String queryString2 = "match $r ($x, $y) isa $type; $r id 'V123'; get;";
+        assertThat(qb.<GetQuery>parse(queryString).execute(), empty());
+        assertThat(qb.<GetQuery>parse(queryString2).execute(), empty());
+    }
+
+    @Test
+    public void whenQueryingForInexistentResourceId_emptyResultReturned(){
+        QueryBuilder qb = testContext.tx().graql().infer(true);
+        String queryString = "match $x has name $y; $x id 'V123'; get;";
+        String queryString2 = "match $x has name $y; $y id 'V123'; get;";
+        String queryString3 = "match $x has name $y via $r; $r id 'V123'; get;";
+        assertThat(qb.<GetQuery>parse(queryString).execute(), empty());
+        assertThat(qb.<GetQuery>parse(queryString2).execute(), empty());
+        assertThat(qb.<GetQuery>parse(queryString3).execute(), empty());
+    }
+
+    @Test
+    public void whenQueryingForInexistentEntityTypeLabel() throws GraqlQueryException{
```

Comment:
Even though it says `throws` at the end, we need to put what should happen in the name of the test so we can see it when it fails on IntelliJ or Jenkins. I usually just put `whenBlah_Throw` because I think that's enough info.